frontend/packages/console-app/src/components/cloud-shell/CloudShellExec.tsx

```diff
@@ -0,0 +1,155 @@
+    this.state = {
+      open: false,
+      error: null,
+    };
+    this.terminal = React.createRef();
+  }
+
+  componentDidMount() {
+    this.connect();
+  }
+
+  componentWillUnmount() {
+    this.ws && this.ws.destroy();
+    delete this.ws;
+  }
+
+  onData = (data: string): void => {
+    this.ws && this.ws.send(`0${Base64.encode(data)}`);
+  };
+
+  connect() {
+    const { container, podname, namespace, shcommand, flags, impersonate } = this.props;
+    const usedClient = flags[FLAGS.OPENSHIFT] ? 'oc' : 'kubectl';
+    const cmd = shcommand || ['sh', '-i', '-c', 'TERM=xterm sh'];
+
+    const params = {
+      ns: namespace,
+      name: podname,
+      path: 'exec',
+      queryParams: {
+        stdout: '1',
+        stdin: '1',
+        stderr: '1',
+        tty: '1',
+        container,
+        command: cmd.map((c) => encodeURIComponent(c)).join('&command='),
+      },
+    };
+
+    if (this.ws) {
+      this.ws.destroy();
+      const currentTerminal = this.terminal.current;
+      currentTerminal && currentTerminal.onConnectionClosed(`connecting to ${container}`);
+    }
+
+    const subprotocols = (impersonate?.subprotocols || []).concat('base64.channel.k8s.io');
+
+    let previous;
+    this.ws = new WSFactory(`${podname}-terminal`, {
+      host: 'auto',
+      reconnect: true,
+      path: resourceURL(PodModel, params),
+      jsonParse: false,
+      subprotocols,
+    })
+      .onmessage((raw) => {
+        const currentTerminal = this.terminal.current;
+        // error channel
+        if (raw[0] === '3') {
+          if (previous.includes(NO_SH)) {
+            currentTerminal.reset();
+            currentTerminal.onConnectionClosed(
+              `This container doesn't have a /bin/sh shell. Try specifying your command in a terminal with:\r\n\r\n ${usedClient} -n ${this.props.namespace} exec ${this.props.podname} -ti <command>`,
+            );
+            this.ws.destroy();
+            previous = '';
+            return;
+          }
+        }
+        const data = Base64.decode(raw.slice(1));
+        currentTerminal && currentTerminal.onDataReceived(data);
+        previous = data;
+      })
+      .onopen(() => {
+        const currentTerminal = this.terminal.current;
+        currentTerminal && currentTerminal.reset();
+        previous = '';
+        this.setState({ open: true, error: null });
+      })
+      .onclose((evt) => {
+        if (!evt || evt.wasClean === true) {
+          return;
+        }
+        const error = evt.reason || 'The terminal connection has closed.';
+        this.setState({ error });
+        this.terminal.current && this.terminal.current.onConnectionClosed(error);
+        this.ws.destroy();
+      }) // eslint-disable-next-line no-console
+      .onerror((evt) => console.error(`WS error?! ${evt}`));
+  }
+
+  render() {
+    const { open, error } = this.state;
+    if (error) {
+      return <StatusBox loadError={error} label="OpenShift command line terminal" />;
```

Comment:
`StatusBox` is incorrectly used here because the `error` needs to confirm to a specific format.
Use `LoadError` instead.
Probably need to also pass `canRetry={false}` because refreshing the page doesn't bring the user back to an open terminal. Ideally we make the the link configurable and restart the terminal instead of refreshing the page (something for future consideration).